scripts/test/Muon/fitting_widgets/basic_fitting/basic_fitting_presenter_test.py

```diff
@@ -0,0 +1,512 @@
+        self.mock_model_number_of_datasets = mock.PropertyMock(return_value=0)
+        type(self.model).number_of_datasets = self.mock_model_number_of_datasets
+        self.presenter.clear_cached_fit_functions = mock.Mock()
+
+        self.presenter.handle_new_data_loaded()
+
+        self.presenter.update_and_reset_all_data.assert_called_with()
+        self.mock_view_plot_guess.assert_called_once_with(False)
+        self.presenter.clear_cached_fit_functions.assert_called_with()
+        self.presenter.reset_tab_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_handle_plot_guess_changed_will_update_plot_guess_using_the_model(self):
+        self.presenter.handle_plot_guess_changed()
+
+        self.mock_view_plot_guess.assert_called_once_with()
+        self.model.update_plot_guess(self.plot_guess)
+
+    def test_that_handle_plot_guess_changed_will_update_plot_guess_using_the_model_if_plot_guess_is_false(self):
+        self.mock_view_plot_guess = mock.PropertyMock(return_value=False)
+        type(self.view).plot_guess = self.mock_view_plot_guess
+
+        self.presenter.handle_plot_guess_changed()
+
+        self.mock_view_plot_guess.assert_called_once_with()
+        self.model.update_plot_guess(False)
+
+    def test_that_handle_undo_fit_clicked_will_attempt_to_reset_the_fit_data_and_notify_that_the_data_has_changed(self):
+        self.presenter.clear_cached_fit_functions = mock.Mock()
+        self.presenter.reset_fit_status_and_chi_squared_information = mock.Mock()
+        self.presenter.update_fit_function_in_view_from_model = mock.Mock()
+
+        self.presenter.handle_undo_fit_clicked()
+
+        self.model.use_cached_function.assert_called_once_with()
+        self.presenter.clear_cached_fit_functions.assert_called_once_with()
+        self.presenter.reset_fit_status_and_chi_squared_information.assert_called_once_with()
+        self.presenter.update_fit_function_in_view_from_model.assert_called_once_with()
+        self.model.update_plot_guess.assert_called_once_with(True)
+        self.presenter.selected_fit_results_changed.notify_subscribers.assert_called_once_with([])
+
+    def test_that_handle_fit_clicked_will_show_a_warning_if_no_data_is_loaded(self):
+        self.mock_model_number_of_datasets = mock.PropertyMock(return_value=0)
+        type(self.model).number_of_datasets = self.mock_model_number_of_datasets
+        self.presenter._perform_fit = mock.Mock()
+
+        self.presenter.handle_fit_clicked()
+
+        self.view.warning_popup.assert_called_once_with("No data selected for fitting.")
+        self.assertTrue(not self.presenter._perform_fit.called)
+
+    def test_that_handle_fit_clicked_will_not_perform_a_fit_if_there_is_no_fit_function(self):
+        self.mock_view_fit_object = mock.PropertyMock(return_value=None)
+        type(self.view).fit_object = self.mock_view_fit_object
+        self.presenter._perform_fit = mock.Mock()
+
+        self.presenter.handle_fit_clicked()
+
```

Comment:
should this not check we have a waarning too?

Reply:
A warning isn't displayed when there is no fit function. Its the same as in the current version